### PR TITLE
Fixes native test and add support for FHIR R5 specification

### PR DESCRIPTION
**Pull request by johnpoth**
Thanks !

### CODE REVIEW

File: extensions/fhir/runtime/pom.xml [hunk 1]
```diff
@@ -70,11 +70,14 @@
...
             <artifactId>commons-logging-jboss-logging</artifactId>
         </dependency>
-
         <dependency>
             <groupId>com.oracle.substratevm</groupId>
             <artifactId>svm</artifactId>
         </dependency>
+        <dependency>
+            <groupId>ca.uhn.hapi.fhir</groupId>
+            <artifactId>hapi-fhir-server</artifactId>
```
Comment (lburgazzoli) [inline extensions/fhir/runtime/pom.xml]: As far as I remember the hapi-fhir-server has a dependency on spring-beans: is that required ? if not we can think about excluding it

Comment (johnpoth) [reply extensions/fhir/runtime/pom.xml]: The Spring dependencies are marked as optional so we don't need to exclude them. I've added exclusions for all other dependencies though. Thanks !

File: integration-tests/fhir/src/main/java/org/apache/camel/quarkus/component/fhir/it/FhirR5Resource.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    @Inject
+    ProducerTemplate producerTemplate;
+
+    @Path("/fhir2json")
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_OCTET_STREAM)
+    public Response fhir2json(String patient) throws Exception {
+        InputStream response = producerTemplate.requestBody("direct:json-to-r5", patient, InputStream.class);
```
[the remark this answers is not in the document]
Comment (johnpoth) [reply integration-tests/fhir/src/main/java/org/apache/camel/quarkus/component/fhir/it/FhirR5Resource.java]: Done. Thanks !

File: integration-tests/fhir/src/main/java/org/apache/camel/quarkus/component/fhir/it/FhirR5Resource.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                .entity(response)
+                .build();
+    }
+
+    @Path("/fhir2xml")
+    @POST
+    @Consumes(MediaType.APPLICATION_XML)
+    @Produces(MediaType.APPLICATION_OCTET_STREAM)
+    public Response fhir2xml(String patient) throws Exception {
+        InputStream response = producerTemplate.requestBody("direct:xml-to-r5", patient, InputStream.class);
```
[the remark this answers is not in the document]
Comment (johnpoth) [reply integration-tests/fhir/src/main/java/org/apache/camel/quarkus/component/fhir/it/FhirR5Resource.java]: Done. Thanks !

**Comment by lburgazzoli** [comment]
ok to test

**Pull request merged**